cmake/thirdparty/SetupHIP.cmake

```diff
@@ -65,14 +62,20 @@ endif()
 
 # hip targets must be global for aliases when created as imported targets
 set(_blt_hip_is_global On)
-if (${BLT_EXPORT_THIRDPARTY})
+if(${BLT_EXPORT_THIRDPARTY})
     set(_blt_hip_is_global Off)
-endif ()
+endif()
 
-blt_import_library(NAME       blt_hip
-                   COMPILE_FLAGS "--rocm-path=${ROCM_PATH}"
-                   EXPORTABLE ${BLT_EXPORT_THIRDPARTY}
-                   GLOBAL ${_blt_hip_is_global})
+if(CMAKE_Fortran_COMPILER_ID STREQUAL "Cray")
+  set(_blt_hip_compile_flags "$<$<COMPILE_LANGUAGE:CXX>:SHELL:--rocm-path=${ROCM_PATH}>")
+else()
+  set(_blt_hip_compile_flags "--rocm-path=${ROCM_PATH}")
+endif()
+
+blt_import_library(NAME          blt_hip
+                   COMPILE_FLAGS ${_blt_hip_compile_flags}
```

Comment:
This is the fix. The rest is just consistent formatting.

Reply:
It turns out that while crayftn < 15 doesn't support the flag crayftn 15 does and we probably actually want to use it when we can?

Reply:
added a check for this too. Thanks @tomstitt !